docs/reference/realm-abi.md

```diff
@@ -0,0 +1,203 @@
+|--------|--------|-------------|--------------|
+| Denom  | string | Coin name   | 1            |
+| Amount | int64  | amount      | 2            |
+ 
+### Fee
+
+| Name      | Type           | Description                       | Field Number |
+|-----------|----------------|-----------------------------------|--------------|
+| GasWanted | int64          | gas requested for the transaction | 1            |
+| GasFee    | [Coin](#coin)  | gas payment fee                   | 2            |
+
+### Signature
+
+| Name      | Type  | Description       | Field Number |
+|-----------|-------|-------------------|--------------|
+| PubKey    | bytes | signer public key | 1            |
+| Signature | bytes | signature         | 2            |
+  
+### Tx
+
+| Name       | Type                             | Description                         | Field Number |
+|------------|----------------------------------|-------------------------------------|--------------|
+| Msgs       | repeated string                  | [Message](#tx-messages) requests    | 1            |
+| Fee        | [Fee](#fee)                      | fee                                 | 2            |
+| Signatures | repeated [Signature](#signature) | signatures                          | 3            |
+| Memo       | string                           | description                         | 4            |
+
+Usually, 2 fees are specified in `Tx`:
+- the gas wanted: the gas requested for `Tx`,
+- the gas fee: the gas payment fee.
+
+## Tx messages
+
+`Tx` `Msgs` being protobuf strings, are encoded using a JSON representation. We describe here
+only messages related to realms.
+
+### MsgAddPkg
+
+Create and initialize a new package.
+
+In JSON, contains a first field `"@type": "/vm.m_addpkg"`.
+
+| Name    | Type                    | Description                    | Field Number |
+|---------|-------------------------|--------------------------------|--------------|
+| Creator | string                  | creator address name or Bech32 | 1            |
+| Package | [Package](#package)     | package definition to load     | 2            |
+| Deposit | repeated [Coin](#coin)  | deposit                        | 3            |
+
+### MsgCall
+
+Execute a Gno statement.
+
+In JSON, contains a first field `"@type": "/vm.m_call"`.
+
+| Name    | Type                   | Description                          | Field Number |
+|---------|------------------------|--------------------------------------|--------------|
+| Caller  | string                 | caller address name or Bech32        | 1            |
+| Send    | repeated [Coin](#coin) | amount to pay                        | 2            |
+| PkgPath | string                 | package path of the function to call | 3            |
+| Func    | string                 | function name                        | 4            |
+| Args    | repeated string        | function arguments                   | 5            |
```

Comment:
I think it's important to make a distinction in that `Send` can take in a single string of multiple coins, like "1ugnot,3leon,5marc", while `Args` is a slice of strings instead of a single one.